wgpu/src/backend/web.rs

```diff
@@ -1113,13 +1113,15 @@ impl crate::context::Context for Context {
         _adapter: &Self::AdapterId,
         _adapter_data: &Self::AdapterData,
     ) -> wgt::SurfaceCapabilities {
+        let format = self.0.get_preferred_canvas_format();
+        let preferred_format = if map_texture_format(wgt::TextureFormat::Bgra8Unorm) == format {
+            wgt::TextureFormat::Bgra8Unorm
+        } else {
+            wgt::TextureFormat::Rgba8Unorm
+        };
         wgt::SurfaceCapabilities {
             // https://gpuweb.github.io/gpuweb/#supported-context-formats
-            formats: vec![
-                wgt::TextureFormat::Bgra8Unorm,
-                wgt::TextureFormat::Rgba8Unorm,
-                wgt::TextureFormat::Rgba16Float,
-            ],
+            formats: vec![preferred_format, wgt::TextureFormat::Rgba16Float],
```

Comment:
that link says though that _by spec_ those formats need to be in there, so leaving one out is actually not allowed 🤔 
Arguably this means that the browser implementation is broken, something that should be logged and linked here.
In any case it would be nice to at least leave a comment on what happened here and why we don't stick to the spec